pgbouncer/datadog_checks/pgbouncer/pgbouncer.py

```diff
@@ -87,49 +95,35 @@ def _collect_stats(self, db, instance_tags):
         metric_scope = [self.STATS_METRICS, self.POOLS_METRICS]
 
         try:
-            cursor = db.cursor()
-            results = None
-
-            for scope in metric_scope:
-
-                metrics = scope['metrics']
-                cols = [m[0] for m in metrics]
-
-                try:
+            with db.cursor(cursor_factory=pgextras.DictCursor) as cursor:
+                for scope in metric_scope:
+                    descriptors = scope['descriptors']
+                    metrics = scope['metrics']
                     query = scope['query']
-                    self.log.debug("Running query: %s" % query)
-                    cursor.execute(query)
-
-                    results = cursor.fetchall()
-
-                except pg.Error as e:
-                    self.log.warning("Not all metrics may be available: %s" % str(e))
-                    continue
-
-                for row in results:
-                    if row[0] == self.DB_NAME:
-                        continue
 
-                    desc = scope['descriptors']
+                    try:
+                        self.log.debug("Running query: %s" % query)
```

Comment:
It may be helpful to have a debug line here that also reports the output of the query to help troubleshoot possible issues in the future. What do you think?

Reply:
I did add something like this but was worried it might be too much output to the log so omitted from the final commit. If you think it's a good idea I'll add it back.